Google.PowerShell/Provider/GoogleCloudStorageProvider.cs

```diff
@@ -859,13 +861,32 @@ protected override object RemoveItemDynamicParameters(string path, bool recurse)
 
         private void RemoveFolder(GcsPath gcsPath, bool recurse)
         {
+            IEnumerable<Object> children = ListChildren(gcsPath, true);
+            string query = $"The item at the path '{gcsPath.ObjectPath}' has children and the Recurse parameter was not specified."
+                + " If you continue, all children will be removed with the item. Are you sure you want to continue?";
+            string caption = "Confirm";
+
+            // Ask the user for permission to delete children of the folder if -Recurse is not supplied.
+            if (children.Any() && !recurse)
+            {
+                if (ShouldContinue(query, caption))
```

Comment:
If we want to force this, we should delete the children first in a loop checked by the four parameter ShouldContinue. This will let the user examine the names of each GcsObject they are deleting, or just say "Yes to All".